File: CSharp/6kyu/ReplaceWithAlphabetPosition.cs
using System.Linq;
using System;
using System.Text;
using System.Text.RegularExpressions;

namespace CSharp._6Kyu
{
    public static partial class Kata6
    {
        public static string AlphabetPosition(string text)
        {
            text = Regex.Replace(text.ToLower(), @"[^a-z]", "");

            return string.Join(" ", text.Select(a => a & 31));
        }

        public static string AlphabetPosition_kata(string text)
        {
            return string.Join(" ", text.ToLower().Where(t => char.IsLetter(t)).Select(c => c - 'a' + 1));
        }

        public static string AlphabetPosition_kata2(string text)
        {
            return string.Join(" ", text.ToLower().Where(c => char.IsLetter(c)).Select(a => a & 31));
        }
    }
}

File: JavaScript/6kyu/ReplaceWithAlphabetPosition.js
function alphabetPosition(text) {
  text = text.toLowerCase().match(/[a-z]/gi);

  if (!text) return "";

  return text.map((c) => "abcdefghijklmnopqrstuvwxyz".indexOf(c) + 1).join(" ");
}

var result = alphabetPosition("123");
console.log(result);
